src/fights.ts

```diff
@@ -606,6 +607,17 @@ class FreeRunFight extends FreeFight {
     while (this.isAvailable()) {
       const initialSpec = undelay(this.options.spec ?? {});
       const constraints = {
+        allowedAction: (action: ActionSource) => {
+          const props = new Map<
+            Item | Familiar | Skill | (Item | Familiar | Skill)[],
+            () => boolean // function that returns true if we should disallow usage of the source while we're reserving embezzler banishers
+          >([
+            [$skill`Snokebomb`, () => get(`_snokebombUsed`) > 0], // We intend to save at least 2 uses for embezzlers, so if we've already used one, disallow usage.
+            [$item`mafia middle finger ring`, () => true],
+          ]);
+          const disallowUsage = props.get(action.source);
+          return !(disallowUsage?.() && get("_garboUsingFreeEmbezzlerBanish", false));
+        },
```

Comment:
You have written this code thrice now; it would be very col if we could get it down to once